### PR TITLE
Reworded 'beyond discovery permissions' wording in "Using RBAC Authorization" page

### LINKED ISSUE #43145: Ambiguity in 'beyond discovery permissions' wording in "Using RBAC Authorization" page
> https://kubernetes.io/docs/reference/access-authn-authz/rbac/#service-account-permissions
> Default RBAC policies grant scoped permissions to control-plane components, nodes, and controllers, but grant no permissions to service accounts outside the kube-system namespace (beyond discovery permissions given to all authenticated users).

> https://kubernetes.io/docs/reference/access-authn-authz/rbac/#upgrading-from-abac
> Default RBAC policies grant scoped permissions to control-plane components, nodes, and controllers, but grant no permissions to service accounts outside the kube-system namespace (beyond discovery permissions given to all authenticated users).

The phrase "beyond discovery permissions given to all authenticated users" in these two sections seems inaccurate. It is quite possible to guess that it references to [API discovery roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#discovery-roles) namely `system:basic-user`, `system:discovery`, `system:public-info-viewer`, but because of "discovery permissions" phrase it seems to only refer to `system:discovery`. It would be better to use different wording, for example:

 - beyond the permissions given by [API discovery roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#discovery-roles)
 - beyond the permissions given by `system:authenticated` and `system:unauthenticated` groups
 - beyond the permissions given by `system:basic-user`, `system:discovery`, `system:public-info-viewer` cluster roles

### PR DESCRIPTION
**Pull request by T-Lakshmi**
Fixes #43145

### CODE REVIEW

**Review by PranitRout07 — COMMENTED** [review]
This sounds good . lgtm

**Comment by T-Lakshmi** [comment]
Thanks @PranitRout07 for review,

@liggitt, @deads2k,
Can you PTAL!!

**Comment by natalisucks** [comment]
@kubernetes/sig-auth-pr-reviews Howdy folks, we'd love a look here to see if this rewording is accurate, thanks!

File: content/en/docs/reference/access-authn-authz/rbac.md [hunk 1]
```diff
@@ -1125,7 +1125,7 @@ Examples:
 
 Default RBAC policies grant scoped permissions to control-plane components, nodes,
 and controllers, but grant *no permissions* to service accounts outside the `kube-system` namespace
-(beyond discovery permissions given to all authenticated users).
+(beyond the permissions given by [API discovery roles](#discovery-roles)).
```
Comment (enj) [inline content/en/docs/reference/access-authn-authz/rbac.md:1128]: Service accounts also have access to API discovery and soon will have access to cluster trust bundles.  There are probably other things I am forgetting.

Comment (T-Lakshmi) [reply content/en/docs/reference/access-authn-authz/rbac.md:1128]: Hi @enj,
Do you wants to hold this PR till the time Service accounts get access to cluster trust bundles and other if any?

Comment (T-Lakshmi) [reply content/en/docs/reference/access-authn-authz/rbac.md:1128]: Hi @enj,
Just pinging as a reminder for above,
Thanks.

**Comment by dipesh-rawat** [comment]
Hello @kubernetes/sig-auth-pr-reviews, we would appreciate your feedback to ensure the accuracy of these proposed changes. Thanks!

**Comment by divya-mohan0209** [comment]
Left [a message](https://kubernetes.slack.com/archives/C0EN96KUY/p1718121409477569) on the SIG Auth Slack channel.

File: content/en/docs/reference/access-authn-authz/rbac.md (hunk 1 above)
Comment (liggitt) [reply content/en/docs/reference/access-authn-authz/rbac.md:1128]: the linked section includes all the discovery-style roles granted to all authenticated users... I would assume any future roles like that would be included there

/lgtm
/approve

**Review by divya-mohan0209 — COMMENTED** [review]
Thanks Jordan for the technical signoff!
/approve

**Pull request merged**